pkg/backup/backup/backup_manager.go

```diff
@@ -14,31 +14,252 @@
+	ns := backup.GetNamespace()
+	name := backup.GetName()
+
+	user, password, reason, err := backuputil.GetTidbUserAndPassword(backup, bm.secretLister)
+	if err != nil {
+		return nil, reason, err
+	}
+
+	storageEnv, reason, err := backuputil.GenerateStorageCertEnv(backup, bm.secretLister)
+	if err != nil {
+		return nil, reason, err
+	}
+
+	// TODO: make pvc request storage size configurable
+	reason, err = bm.ensureBackupPVCExist(backup)
+	if err != nil {
+		return nil, reason, err
+	}
+
+	args := []string{
+		"backup",
+		fmt.Sprintf("--namespace=%s", ns),
+		fmt.Sprintf("--tidbcluster=%s", backup.Spec.Cluster),
+		fmt.Sprintf("--backupName=%s", name),
+		fmt.Sprintf("--tidbservice=%s", controller.TiDBMemberName(backup.Spec.Cluster)),
+		fmt.Sprintf("--storageType=%s", backup.Spec.StorageType),
+		fmt.Sprintf("--password=%s", password),
+		fmt.Sprintf("--user=%s", user),
+	}
+
+	backupLabel := label.NewBackup().Instance(backup.Spec.Cluster).BackupJob().Backup(name)
+
+	// TODO: need add ResourceRequirement for backup job
+	podSpec := &corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: backupLabel.Labels(),
+		},
+		Spec: corev1.PodSpec{
+			ServiceAccountName: constants.DefaultServiceAccountName,
+			Containers: []corev1.Container{
+				{
+					Name:            label.BackupJobLabelVal,
+					Image:           controller.TidbBackupManagerImage,
+					Args:            args,
+					ImagePullPolicy: corev1.PullAlways,
+					VolumeMounts: []corev1.VolumeMount{
+						{Name: label.BackupJobLabelVal, MountPath: constants.BackupRootPath},
+					},
+					Env: storageEnv,
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyNever,
+			Volumes: []corev1.Volume{
+				{
+					Name: label.BackupJobLabelVal,
+					VolumeSource: corev1.VolumeSource{
+						PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+							ClaimName: backup.GetBackupPVCName(),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      backup.GetBackupJobName(),
+			Namespace: ns,
+			Labels:    backupLabel,
+			OwnerReferences: []metav1.OwnerReference{
+				controller.GetBackupOwnerRef(backup),
+			},
+		},
+		Spec: batchv1.JobSpec{
+			BackoffLimit: controller.Int32Ptr(0),
+			Template:     *podSpec,
+		},
+	}
+
+	return job, "", nil
+}
+
+func (bm *backupManager) ensureBackupPVCExist(backup *v1alpha1.Backup) (string, error) {
+	ns := backup.GetNamespace()
+	name := backup.GetName()
+
+	storageSize := constants.DefaultStorageSize
+	if backup.Spec.StorageSize != "" {
+		storageSize = backup.Spec.StorageSize
+	}
+	rs, err := resource.ParseQuantity(storageSize)
+	if err != nil {
+		errMsg := fmt.Errorf("backup %s/%s parse storage size %s failed, err: %v", ns, name, constants.DefaultStorageSize, err)
+		return "ParseStorageSizeFailed", errMsg
+	}
+	backupPVCName := backup.GetBackupPVCName()
+	_, err = bm.pvcLister.PersistentVolumeClaims(ns).Get(backupPVCName)
+	if err != nil {
```

Comment:
here err should be IsNotFound?

Reply:
I have added a comment below to explain this, but it is better to use `IsNotFound` explicitly.